core/math/rect2.h

```diff
@@ -152,14 +152,23 @@ struct Rect2 {
 		return new_rect;
 	};
 	inline bool has_point(const Point2 &p_point) const {
-		if (p_point.x < position.x)
+		Point2 _position;
+		Size2 _size;
+
+		_position.x = MIN(position.x, position.x + size.x);
```

Comment:
All this code could simply be replaced by a call to `Rect2.abs()` on a copy of the rect. And then the remaining code could be left intact if ran on the copy instead.

Reply:
@groud I've simplified the code with `abs()` and the not modifying the negative size ( unless there is no way to tell it the rect is flipped )